src/main/java/com/emc/mongoose/base/load/step/file/FileManagerImpl.java

```diff
@@ -56,20 +56,33 @@ public final String newTmpFileName() throws IOException {
 			if (buffSize > 0) {
 				final ByteBuffer bb = ByteBuffer.allocate(buffSize);
 				int doneSize = 0;
-				int n;
+				int readBytesTotal;
+				int newLineCharacterCode = '\n';  //  == 10
+				byte[] resultingBuffer = new byte[0];
 				while (doneSize < buffSize) {
-					n = fileChannel.read(bb);
-					if (n < 0) {
+					readBytesTotal = fileChannel.read(bb);
+					int lastReadItemIndex = readBytesTotal - 1;
+					if (readBytesTotal < 0) {
 						// unexpected but possible: the file is shorter than was estimated before
 						final byte[] buff = new byte[bb.position()];
 						bb.rewind();
 						bb.get(buff);
 						return buff;
+					} else if (bb.get(lastReadItemIndex) != newLineCharacterCode) {
+						int lastNewLineCharacterIndex = lastReadItemIndex;
+						while (bb.get(lastNewLineCharacterIndex) != newLineCharacterCode) {
+							lastNewLineCharacterIndex--;
+						}
+						resultingBuffer = new byte[lastNewLineCharacterIndex + 1];
+						bb.rewind();
+						bb.get(resultingBuffer, 0, lastNewLineCharacterIndex + 1);
+						doneSize += readBytesTotal;
```

Comment:
Should `readBytesTotal` be changed in case Mongoose went backwards to last new line symbol? I assume `doneSize` is the amount of processed bytes from the local tmp file.

Reply:
readBytesTotal reflects how many bytes we've read. It should not go down after we reject some bytes

Reply:
I'll add a comment on that